spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/interpreter/fixtures/StreamingFixture.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.conformance.interpreter.fixtures
+
+import org.apache.spark.sql.{DataFrame, Row}
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.execution.streaming.MemoryStream
+import org.apache.spark.sql.streaming.Trigger
+import org.scalatest.FunSuite
+import org.scalatest.mockito.MockitoSugar
+import za.co.absa.enceladus.conformance.{ConfCmdConfig, HyperConformance}
+import za.co.absa.enceladus.conformance.interpreter.FeatureSwitches
+import za.co.absa.enceladus.conformance.streaming.{InfoDateFactory, InfoDateLiteralFactory}
+import za.co.absa.enceladus.dao.MenasDAO
+import za.co.absa.enceladus.model.Dataset
+import za.co.absa.enceladus.utils.testUtils.SparkTestBase
+
+trait StreamingFixture extends FunSuite with SparkTestBase with MockitoSugar {
+  implicit val menasBaseUrls: List[String] = List()
+  implicit val cmd: ConfCmdConfig = ConfCmdConfig.apply(reportVersion = Some(1))
+  implicit val infoDateFactory: InfoDateFactory = new InfoDateLiteralFactory("2020-05-23")
+  private val processingTime = 1500
+
+  protected def testHyperConformance(input: DataFrame,
+                                     sinkTableName: String,
+                                     dataset: Dataset,
+                                     catalystWorkaround: Boolean = true)
+                                    (implicit menasDAO: MenasDAO): DataFrame = {
+    implicit val featureSwitches: FeatureSwitches = FeatureSwitches()
+      .setExperimentalMappingRuleEnabled(false)
+      .setCatalystWorkaroundEnabled(catalystWorkaround)
+      .setControlFrameworkEnabled(false)
+
+    val memoryStream = new MemoryStream[Row](1, spark.sqlContext)(RowEncoder(input.schema))
+    val hyperConformance = new HyperConformance()
+    val source: DataFrame = memoryStream.toDF()
+    val conformed: DataFrame = hyperConformance.applyConformanceTransformations(source, dataset)
+    val sink = conformed
+      .writeStream
+      .trigger(Trigger.ProcessingTime(processingTime))
+      .queryName(sinkTableName)
+      .outputMode("append")
+      .format("memory")
+      .start()
+
+    input.collect().foreach(e => {
+      memoryStream.addData(e)
+      Thread.sleep(100)
```

Comment:
Is there a way it can be done so it won't need to rely on timing. Since if the timing is involved we might see the test fail non-deterministically since it is going to be run on machines that have very different specs.

Maybe you can wait for a micro-batch to be processed before sending another one by doing `select` and `count` on the memory table?

Reply:
About timing, I don't say the tests couldn't fail, but it's not likely. As @kevinwallimann suggested, the timeout between the insertion better simulates a more realistic use case and since inserting everything would take 2 seconds, the total waited time is 6 seconds and I can incrase it. 

The other discussion we had about not using `awaitTermination` and using `processAllAvailable` is that using the second one makes testing with the workaround disabled run an unreasonable amount of time.

Reply:
Sorry, I wasn't very constructive since I don't have a solution to this. Structured Streaming jobs is a new topic for me as well.
Not sure if it can be done without relying on the timer. But if it could be done, such a solution would be preferable.

Reply:
Thanks a lot, @kevinwallimann for the insight and for the proposed solution!

@AdrianOlosutean, I'm still not 100% sure if we should have a [uncommented] test that runs forever if the workaround is turned off and is interrupted by a timer. Let me know what you think. My current concern is that having such a test would bring more risks than benefits. We need to ensure that the processing interval does interrupt the query even if it is stuck inside the Catalyst layer.
Having a commented test that hangs without the workaround can be beneficial to test new versions of Spark. And since it is commented, it doesn't take time to run on normal builds :)

Reply:
For me it seems to that the commented test would be the more sensible approach. Very interesting, obviously the solution proposed by @kevinwallimann is the rigurous one and I guess it should be used, thanks.